crates/pathfinder/src/sequencer/request.rs

```diff
@@ -33,3 +33,273 @@ impl From<rpc::Call> for Call {
+    }
+
+    #[derive(Copy, Clone, Debug, serde::Deserialize, serde::Serialize, PartialEq, Hash, Eq)]
+    #[serde(deny_unknown_fields)]
+    pub enum EntryPointType {
+        #[serde(rename = "EXTERNAL")]
+        External,
+        #[serde(rename = "L1_HANDLER")]
+        L1Handler,
+        #[serde(rename = "CONSTRUCTOR")]
+        Constructor,
+    }
+
+    impl fmt::Display for EntryPointType {
+        fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+            use EntryPointType::*;
+            f.pad(match self {
+                External => "EXTERNAL",
+                L1Handler => "L1_HANDLER",
+                Constructor => "CONSTRUCTOR",
+            })
+        }
+    }
+
+    #[derive(Clone, Debug, serde::Deserialize, serde::Serialize)]
+    #[serde(deny_unknown_fields)]
+    pub struct SelectorAndOffset {
+        pub selector: EntryPoint,
+        pub offset: ByteCodeOffset,
+    }
+
+    // It's important that this is ordered alphabetically because the fields need to be in
+    // sorted order for the keccak hashed representation.
+    #[derive(serde::Deserialize, serde::Serialize)]
+    #[serde(deny_unknown_fields)]
+    pub struct Program<'a> {
+        #[serde(skip_serializing_if = "Vec::is_empty", default)]
+        pub attributes: Vec<serde_json::Value>,
+
+        #[serde(borrow)]
+        pub builtins: Vec<Cow<'a, str>>,
+
+        #[serde(borrow)]
+        pub data: Vec<Cow<'a, str>>,
+
+        #[serde(borrow)]
+        pub debug_info: Option<&'a serde_json::value::RawValue>,
+
+        // Important that this is ordered by the numeric keys, not lexicographically
+        pub hints: BTreeMap<u64, Vec<serde_json::Value>>,
+
+        pub identifiers: serde_json::Value,
+
+        #[serde(borrow)]
+        pub main_scope: Cow<'a, str>,
+
+        // Unlike most other integers, this one is hex string. We don't need to interpret it,
+        // it just needs to be part of the hashed output.
+        #[serde(borrow)]
+        pub prime: Cow<'a, str>,
+
+        pub reference_manager: serde_json::Value,
+    }
+
+    #[cfg(test)]
+    mod roundtrip_tests {
+        // FIXME: we should have many test cases utilizing this.
+        #[allow(unused)]
+        fn roundtrips<'a, T>(input: &'a str)
+        where
+            T: serde::Deserialize<'a> + serde::Serialize,
+        {
+            use crate::state::contract_hash::PythonDefaultFormatter;
+
+            let parsed: T = serde_json::from_str(input).unwrap();
+            let mut ser =
+                serde_json::Serializer::with_formatter(Vec::new(), PythonDefaultFormatter);
+            parsed.serialize(&mut ser).unwrap();
+            let bytes = ser.into_inner();
+            let output = std::str::from_utf8(&bytes).expect("serde does this unchecked");
+
+            // these need to be byte for byte equal because we hash this
+            assert_eq!(input, output);
+        }
+    }
```

Comment:
```suggestion
```

Could just delete since it hasn't been needed.

Reply:
Done.